<a href="https://colab.research.google.com/github/TristanConant/shroom_class/blob/init/project_2_intro.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

In [6]:
file_url = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vQ94PFNlOYMndbOU8lOH9YTaPKyngt0V2KDC47hQnGm9fvdbtXa3OXhu3AKnDzzBYEQlSw2UxS7Q7XI/pub?output=csv'
df = pd.read_csv(file_url, header=None)
df.columns = ['edible', 'cap-shape', 'cap-surface', 'cap-color', 'bruises', 'odor', 'gill-attachment', 'gill-spacing', 'gill-size', 'gill-color', 'stalk-shape', 'stalk-root', 'stalk-surface-above-ring', 'stalk-surface-below-ring', 'stalk-color-above-ring', 'stalk-color-below-ring', 'veil-type', 'veil-color', 'ring-number', 'ring-type', 'spore-print-color', 'population', 'habitat']
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8124 entries, 0 to 8123
Data columns (total 23 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   edible                    8124 non-null   object
 1   cap-shape                 8124 non-null   object
 2   cap-surface               8124 non-null   object
 3   cap-color                 8124 non-null   object
 4   bruises                   8124 non-null   object
 5   odor                      8124 non-null   object
 6   gill-attachment           8124 non-null   object
 7   gill-spacing              8124 non-null   object
 8   gill-size                 8124 non-null   object
 9   gill-color                8124 non-null   object
 10  stalk-shape               8124 non-null   object
 11  stalk-root                8124 non-null   object
 12  stalk-surface-above-ring  8124 non-null   object
 13  stalk-surface-below-ring  8124 non-null   object
 14  stalk-color-above-ring  

1) Source of data<br>
[Data source](https://archive-beta.ics.uci.edu/ml/datasets/mushroom#Descriptive)<br>
2) Brief description of data<br>
Mushrooms described in terms of physical characteristics.<br>
3) What is the target?<br>
poisonous or edible<br>
4) Is this a classification or regression problem?<br>
classification<br>
5) How many features?<br>
22<br>
6) How many rows of data.<br>
8123<br>
7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?<br>
I believe I might have some trouble with missing data.

In [8]:
file_url = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vT1Jag_dcBqT63ebPFDnrsznyatSXKEc2RdDmqkgutwMQ-TJe2JVBNXPwL4-3QTo_70FJ3TBz62DCTC/pub?output=csv'
df = pd.read_csv(file_url, header=None)
df.columns = ['erythema', 'scaling', 'definite borders', 'itching', 'koebner phenomenon', 'polygonal papules', 'follicular papules', 'oral mucosal involvement', 'knee and elbow involvement', 'scalp involvement', 'family history, (0 or 1)', 'Age (linear)', 'melanin incontinence', 'eosinophils in the infiltrate', 'PNL infiltrate', 'fibrosis of the papillary dermis', 'exocytosis', 'acanthosis', 'hyperkeratosis', 'parakeratosis', 'clubbing of the rete ridges', 'elongation of the rete ridges', 'thinning of the suprapapillary epidermis', 'spongiform pustule', 'munro microabcess', 'focal hypergranulosis', 'disappearance of the granular layer', 'vacuolisation and damage of basal layer', 'spongiosis', 'saw-tooth appearance of retes', 'follicular horn plug', 'perifollicular parakeratosis', 'Clinical Attributes', 'inflammatory monoluclear inflitrate', 'band-like infiltrate']
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 366 entries, 0 to 365
Data columns (total 35 columns):
 #   Column                                    Non-Null Count  Dtype 
---  ------                                    --------------  ----- 
 0   erythema                                  366 non-null    int64 
 1   scaling                                   366 non-null    int64 
 2   definite borders                          366 non-null    int64 
 3   itching                                   366 non-null    int64 
 4   koebner phenomenon                        366 non-null    int64 
 5   polygonal papules                         366 non-null    int64 
 6   follicular papules                        366 non-null    int64 
 7   oral mucosal involvement                  366 non-null    int64 
 8   knee and elbow involvement                366 non-null    int64 
 9   scalp involvement                         366 non-null    int64 
 10  family history, (0 or 1)                  366 non-

1) Source of data<br>
[Data source](https://archive-beta.ics.uci.edu/ml/datasets/dermatology#Descriptive)<br>
2) Brief description of data<br>
This dataset is to determine the type of Eryhemato-Squamous Disease.<br>
3) What is the target?<br>
Diagnosis of erythemato-squamous diseases<br>
4) Is this a classification or regression problem?<br>
classification<br>
5) How many features?<br>
34<br>
6) How many rows of data.<br>
366<br>
7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?<br>
Eliminating unnecessary columns.